src/NLPModels.jl

```diff
@@ -272,6 +281,25 @@ Evaluate ``∇c(x)v``, the Jacobian-vector product at `x` in place.
 jprod!(::AbstractNLPModel, ::AbstractVector, ::AbstractVector, ::AbstractVector) =
   throw(NotImplementedError("jprod!"))
 
+"""
+    Jv = jprod!(nlp, rows, cols, vals, v, Jv)
+
+Evaluate ``∇c(x)v``, the Jacobian-vector product, where the Jacobian is given by
+`(rows, cols, vals)` in triplet format.
+"""
+function jprod!(nlp::AbstractNLPModel, rows::AbstractVector{<: Integer}, cols::AbstractVector{<: Integer}, vals::AbstractVector, v::AbstractVector, Jv::AbstractVector)
+  increment!(nlp, :neval_jprod)
+  coo_prod!(nlp.meta.ncon, nlp.meta.nnzj, rows, cols, vals, v, Jv)
```

Comment:
Shouldn't we deduce `nnzj` from the common length of `rows`, `cols` and `vals`? I think the user should pass arrays (or views) of the correct size.

Reply:
It's cheaper to use `nlp.meta.nnzj` than `length`, and if the user is passing vectors of the correct size, then it doesn't change the result.